hypervisor/acpi_parser/dmar_parse.c

```diff
@@ -204,11 +205,14 @@ handle_one_drhd(struct acpi_dmar_hardware_unit *acpi_drhd,
 
 		consumed = handle_dmar_devscope(dev_scope, cp, remaining);
 
-		if (((drhd->segment << 16U) |
-		     (dev_scope->bus << 8U) |
-		     dev_scope->devfun) == CONFIG_GPU_SBDF) {
-			ASSERT(dev_count == 1, "no dedicated iommu for gpu");
-			drhd->ignore = true;
+		/* Disable GPU IOMMU due to gvt-d hasn’t been enabled on APL yet. */
+		if (is_apl_platform()) {
+			if (((drhd->segment << 16U) |
+		     	     (dev_scope->bus << 8U) |
+		     	     dev_scope->devfun) == CONFIG_GPU_SBDF) {
+				ASSERT(dev_count == 1, "no dedicated iommu for gpu");
```

Comment:
If this is never to happen case, we need to use @pre  to make sure it never happen for FuSa.
If this may happen, we need gracefully handle it. ASSERT is not graceful.
In this case, this is the old code, and I believe we never hit here, can you check with Binbin to fix it after this patch

Reply:
OK. I will check this with binbin.